### PR TITLE
Dev - Adding FastQC to the pipeline, with integration into MultiQC

### PR DESCRIPTION
**Pull request by heylf**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/fastqc.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      .map { ch -> [ ch[0], ch[1] ] }
+      .set { ch_fastq }
+
+  /*
+   * FastQ QC using FASTQC
+   */
+  fastqc_zip     = Channel.empty()
+  fastqc_html    = Channel.empty()
+  fastqc_multiqc = Channel.empty()
+  fastqc_version = Channel.empty()
```
Comment (grst) [inline subworkflows/local/fastqc.nf:24]: Is this necessary?

**Review by grst — APPROVED** [review]
Looks great! Just two small code questions...

**Comment by apeltzer** [comment]
@nf-core-bot fix linting

File: subworkflows/local/fastqc.nf (hunk 1 above)
Comment (heylf) [reply subworkflows/local/fastqc.nf]: no, some are obsolete, which came from old code.

File: workflows/scrnaseq.nf [hunk 2]
```diff
@@ -105,6 +106,14 @@ workflow SCRNASEQ {
 
     ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
 
+    // Run FastQC
+    ch_multiqc_fastqc = Channel.empty()
+    if (!params.skip_fastqc){
+      FASTQC_CHECK ( ch_fastq )
+      ch_versions = ch_versions.mix(FASTQC_CHECK.out.fastqc_version.first().ifEmpty(null))
```
[the remark this answers is not in the document]
Comment (heylf) [reply workflows/scrnaseq.nf]: I have removed it

File: subworkflows/local/fastqc.nf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+//
+// Check input samplesheet and get read channels
+//
+
+//TODO --> add skip_fastqc to params
```
[the remark this answers is not in the document]
Comment (heylf) [reply subworkflows/local/fastqc.nf]: Yes!

**Pull request merged**